relnotes/pkg-iteration.feature.md

```diff
@@ -0,0 +1,42 @@
+## The package iteration can now be specified via `PKGITERATION`
+
+Package versions consist of two components: the _upstream version_, and
+the _package iteration_ (corresponding to the `debian_revision` where
+deb packages are concerned).
+
+Previously the package iteration was hardcoded to match the distro code
+(e.g. `xenial`), disallowing typical patterns referencing the packager
+and changes to the packaging rather than the upstream project.  For
+example:
+
+* Debian packages use a simple number for the `debian_revision` (1, 2,
+  3, ...), resetting to 1 when the upstream version changes
+
+* Ubuntu packages typically use an `XubuntuY` format where `X` is the
+  numerical revision of the Debian package that has been forked (or `0`
+  if there is no such package), and `Y` is the Ubuntu package revision
+  (hence Ubuntu package versions like `2.7.0-2ubuntu3.1`, where `2.7.0`
+  is the upstream version and `2ubuntu3.1` means Ubuntu revision 3.1 of
+  Debian package revision 2)
+
+* some packagers like to include both revision information and the build
+  or target distro: e.g. packages from the git PPA have revisions such
+  as `0ppa1~ubuntu16.04.1` (note also the leading 0 indicating that this
+  is a new package rather than a fork of a Debian package).
+
+The new `PKGITERATION` variable allows the package iteration to be set
+to whatever is wanted, whether via `Config.mak` or the command line.
+It defaults to the distro code, meaning nothing will change for existing
+package definitions, but can now be customized where this is appropriate
+(e.g. for packages of 3rd-party projects).
+
+Note that if you want to include the distro code in a customized package
+iteration, this will have to be included manually.  For example,
+`PKGITERATION=3~"$(lsb_release -cs)"` will produce iterations such as
+`3~trusty`, `3~xenial`, etc. depending on the distro used to build the
+package.
+
+See https://www.debian.org/doc/debian-policy/#version for the official
+Debian package-versioning spec, and
+https://askubuntu.com/questions/620533/how-does-ubuntu-name-packages
+for an explanation of how Ubuntu specifies the `debian_revision`.
```

Comment:
OK, I like how now it doesn't feel like the only option is the ubuntu scheme, but I still feel this is too verbose and out of place.

First, as a user, if I don't care about this new feature, I don't want to have to scroll 3 pages to have a glimpse of what the next new feature is. This got even longer as before, so more distracting.

Then, is too detailed, to much history. A changelog should be short and to the point. Longer and more descriptive documentation should go in the documentation proper, the readme or whatever. I thought of moving it to the `PKGITERATION` or `iteration` docs in the README. But the README right now is working more like a reference, look at the other variable definitions, they are super short and to the point.

For me it would be super weird that for `PKGVERSION` you'll get:
```rest
* ``PKGVERSION`` is the version to be used when creating packages. It's
   obtained via the ``VERSION`` variable by default.
```
and for `PKGITERATION` a 6 paragraphs-long description with many options of different schemes used by different projects.

For me this kind of description belongs to a tutorial, not to a reference or a changelog, it just dissonant with the rest of the project's docs and changelogs. This project assumes knowledge about packaging, and when you need detailed information, it just points you to another resource with a link to the main source. For me in this case the main source is the Debian docs.

For me this changelog and the README should just say:
```rest
Allows setting the package *iteration* (forwarded to `fpm` as `--iteration`, known as `debian_revision` [in Debian](https://www.debian.org/doc/debian-policy/#version)).
```
(in the changelog it would be nice to also mention that before it was hardcoded to `$(lsb_release -cs)`).

I'm sorry, I know you spend some time looking for these sources and it took time and effort to write, and I know is useful information, but again, I think this is not the right place to transmit this information, and for the changelog is counter-productive to be so verbose and for the README it just doesn't go with the style of the rest of the README.

Reply:
No worries, it's a good point about keeping things simple and brief.  I'll cut things down for this PR and we can follow up with the detailed info in a wiki page or somewhere else.